### PR TITLE
Carouselbranch

### PR DESCRIPTION
**Pull request by silkyh13**

### CODE REVIEW

File: client/components/app.jsx [hunk 1]
```diff
@@ -6,31 +6,62 @@ import CarouselPic from './carousel.jsx';
...
+      show: false,
+      like: false,
+      productId: null
     };
     this.nextPicture = this.nextPicture.bind(this);
     this.prevPicture = this.prevPicture.bind(this);
     this.selectedPic = this.selectedPic.bind(this);
     this.toggleModal = this.toggleModal.bind(this);
+    this.toggleHeart = this.toggleHeart.bind(this);
+    this.componentDidMount = this.componentDidMount.bind(this);
```
Comment (mflinn92) [inline client/components/app.jsx]: I don't know if you need this line, I think component did mount automatically has the correct binding.

File: client/components/app.jsx [hunk 2]
```diff
@@ -6,31 +6,62 @@ import CarouselPic from './carousel.jsx';
 import DisplayPic from './display.jsx';
 import LeftArrow from './leftArrow.jsx';
 import RightArrow from './rightArrow.jsx';
+import HeartButton from './heartButton.jsx';
```
Comment (therobinkim) [inline client/components/app.jsx]: Please see https://github.com/airbnb/javascript/blob/master/react/README.md#naming re: file names (and also configure ESLint for Airbnb's style guide asap).

File: client/components/app.jsx [hunk 3]
```diff
@@ -6,31 +6,62 @@ import CarouselPic from './carousel.jsx';
...
+    this.componentDidMount = this.componentDidMount.bind(this);
+
   }
+
   toggleModal (event) {
     this.setState({ show: !this.state.show})
   }
 
+  async toggleHeart (event) {
+    await this.setState({ like: !this.state.like});
```
Comment (therobinkim) [inline client/components/app.jsx]: Is `this.setState` is `await`-able?

File: client/components/app.jsx [hunk 4]
```diff
@@ -6,31 +6,62 @@ import CarouselPic from './carousel.jsx';
...
+    axios.put('/products', {
+      productId: this.state.productId,
```
Comment (therobinkim) [inline client/components/app.jsx]: You should specify the ID number in the URL if you are trying to be RESTful.

File: client/components/app.jsx [hunk 5]
```diff
@@ -6,31 +6,62 @@ import CarouselPic from './carousel.jsx';
...
+        console.log(err);
+      });
+  }
+
+  componentDidMount() {
+    var randomNum = (max) => {
+      return Math.floor(Math.random() * max) + 1;
+    };
+    var productId = randomNum(5);
+    this.setState({productId: productId});
```
Comment (therobinkim) [inline client/components/app.jsx]: `this.setState({productId: productId});` => `this.setState({ productId });`

File: client/components/app.jsx [hunk 6]
```diff
@@ -6,31 +6,62 @@ import CarouselPic from './carousel.jsx';
...
+      .catch(error => {
+        console.log(err);
+      });
+  }
+
+  componentDidMount() {
+    var randomNum = (max) => {
+      return Math.floor(Math.random() * max) + 1;
+    };
+    var productId = randomNum(5);
```
Comment (therobinkim) [inline client/components/app.jsx]: You should discuss w/ your team how each module is determining for which productId you are rendering information.

File: server/index.js [hunk 7]
```diff
@@ -8,18 +8,34 @@ app.use(parser.json());
...
+//add product name, url, username, like
 app.post('/products', function (req, res) {
-  models.saveProduct(req.body.productId, req.body.productItem, req.body.pictureUrl);
+  models.saveProduct(req.body.productId, req.body.productItem, req.body.pictureUrl, req.body.like);
+  res.end('done');
+});
+//update like of productId
+app.put('/products', function (req, res) {
+  models.updateProduct(req.body.productId, req.body.like);
   res.end('done');
```
Comment (therobinkim) [inline server/index.js]: This is an asynchronous race condition, and the response is ended with `done` before the database has been successfully updated (or perhaps errors out); we should respond with an error status code/message if an error occurs.

File: server/models/index.js [hunk 8]
```diff
@@ -41,21 +43,41 @@ let saveWishlist = async (products, username) => {
...
+    callback(err, docs);
+  });
 };
 //get wishlist by username
-let getWishlistByUsername = async (username) => {
-  return await MyWishlistModel.find({username: username});
+let getWishlistByUsername = (username, callback) => {
+  MyWishlistModel.find({username: username}, (err, docs) => {
+    callback(err, docs);
+  });
```
Comment (therobinkim) [inline server/models/index.js]:
```
MyWishlistModel.find({username: username}, (err, docs) => {
  callback(err, docs);
});
```

can be abbreviated to `MyWishlistModel.find({username}, callback);`

File: server/models/index.js [hunk 9]
```diff
@@ -41,21 +43,41 @@ let saveWishlist = async (products, username) => {
...
+};
+//update product when liked
+let updateProduct = (productId, like, callback) => {
+
+  MyProductsModel.updateOne({productId: productId}, {like: like}, (err, docs) => {
+    try {
+      return docs;
```
Comment (therobinkim) [inline server/models/index.js]: This `return` statement doesn't do anything. We should be invoking the `callback`.

File: client/components/app.jsx (hunk 3 above)
Comment (silkyh13) [reply client/components/app.jsx]: Yes, it is await-able.

Comment (therobinkim) [reply client/components/app.jsx]: @silkyh13 Ah, poorly worded question; everything in JavaScript is `await`able.

What benefit does `await`ing `this.setState` offer? I am not aware of a Promise API for `this.setState`.

Comment (silkyh13) [reply client/components/app.jsx]: I tried it with both with and without await and it works with await. When await is not there, my put request does not update properly.

**Pull request merged**